core/src/main/java/nl/nn/adapterframework/webcontrol/api/TransactionalStorage.java

```diff
@@ -85,21 +93,29 @@ public Response browseReceiverMessage(
 			throw new ApiException("Adapter not found!");
 		}
 
-		Receiver<?> receiver = adapter.getReceiverByName(receiverName);
-		if(receiver == null) {
-			throw new ApiException("Receiver ["+receiverName+"] not found!");
+		String message = null;
+		IMessageBrowser<?> storage = null;
+		messageId = Misc.urlDecode(messageId);
+
+		if(STORAGESOURCE_TYPE_PIPE.equals(storageSource)) {
+			MessageSendingPipe pipe = (MessageSendingPipe) adapter.getPipeLine().getPipe(storageSourceName);
+			if(pipe == null) {
+				throw new ApiException("Pipe ["+storageSourceName+"] not found!");
+			}
+			storage = pipe.getMessageLog();
+			message = getMessage(storage, messageId);
+		} else {
+			Receiver<?> receiver = adapter.getReceiverByName(storageSourceName);
+			if(receiver == null) {
+				throw new ApiException("Receiver ["+storageSourceName+"] not found!");
+			}
+			storage = receiver.getMessageBrowser(ProcessState.getProcessStateFromName(processState));
+			message = getMessage(storage, receiver.getListener(), messageId);
 		}
 
-		IMessageBrowser<?> storage = receiver.getMessageBrowser(ProcessState.getProcessStateFromName(processState));
 		try {
-			// messageId is double URLEncoded, because it can contain '/' in ExchangeMailListener
-			messageId = Misc.urlDecode(messageId);
```

Comment:
What happened to this?

Reply:
I reordered some of the lines here. This line is moved to line 98.

Reply:
OK, but please keep the comment about the double URLEncoding